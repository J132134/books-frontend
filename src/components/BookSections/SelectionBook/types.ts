import { DisplayType, BookItem } from 'src/types/sections';

export interface SelectionBookCommonProps {
  type: DisplayType;
  genre: string;
  slug: string;
  items: BookItem[]; // Fixme Md 타입 말고 comics UserPreferredSection 타입이 API 결과로 오는데 이 부분 확인해야 함
}

export type SelectionBookListProps = SelectionBookCommonProps;
