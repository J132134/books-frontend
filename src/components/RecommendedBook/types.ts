import { HotRelease, TodayRecommendation } from 'src/types/sections';

interface CommonProps {
  theme: 'dark' | 'white';
  slug: string;
  genre: string;
  title: string;
}

interface TodayRecommendationProps {
  type: 'TodayRecommendation';
  items: TodayRecommendation[];
}

interface HotReleaseProps {
  type: 'HotRelease';
  items: HotRelease[];
}

export type RecommendedBookProps = CommonProps & (TodayRecommendationProps | HotReleaseProps);
