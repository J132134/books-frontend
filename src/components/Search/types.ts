import { Author, SearchBookDetail } from 'src/types/searchResults';

export interface SearchResult {
  books: SearchBookDetail[];
  authors: Author[];
}
