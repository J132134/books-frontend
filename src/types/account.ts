export interface LoggedUser {
  email: string;
  id: string;
  idx: number;
  is_verified_adult: boolean;
}
