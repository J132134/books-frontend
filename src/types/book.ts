export enum AuthorRole {
  AUTHOR = 'author',
  COMIC_AUTHOR = 'comic_author',
  STORY_WRITER = 'story_writer',
  TRANSLATOR = 'translator',
  ILLUSTRATOR = 'illustrator',
  ORIGINAL_AUTHOR = 'original_author',
  AUTHOR_PHOTO = 'author_photo',
  PLANNER = 'planner',
  BIBLIOGRAPHICAL_INTRODUCTION = 'bibliographical_introduction',
  COMPILER = 'compiler',
  COMMENTATOR = 'commentator',
  EDITOR = 'editor',
  SUPERVISE = 'supervise',
  PERFORMER = 'performer',
  ORIGINAL_ILLUSTRATOR = 'original_illustrator',
}

export interface Title {
  main: string;
  prefix?: string;
  sub?: string;
}
export interface Category {
  id: number;
  is_series_category: boolean;
  name: string;
  genre: string;
  sub_genre: string;
}

export interface File {
  character_count: number;
  format: string;
  is_comic: boolean;
  is_comic_hd: boolean;
  is_drm_free: boolean;
  is_manga: boolean;
  is_webtoon: boolean;
  size: number;
}

export interface BuyInfo {
  discount_percentage: number;
  price: number;
  regular_price: number;
}

export interface SeriesBuyInfo {
  free_book_count: number;
  price: number;
  regular_price: number;
  total_book_count: number;
  discount_percentage: number;
}

export interface RentBuyInfo {
  total_book_count: number;
  free_book_count: number;
  regular_price: number;
  rent_price: number;
  rent_days: number;
  discount_percentage: number;
}

export interface SeriesPriceInfo {
  buy: SeriesBuyInfo;
  rent?: RentBuyInfo;
}

export interface PaperBuyInfo {
  price: number;
}

export interface RentInfo {
  regular_price: number;
  price: number;
  rent_days: number;
  discount_percentage: number;
}

export interface PointBackInfo {
  pointback_amount: number;
  point_duration: number;
}

export interface PriceInfo {
  buy?: BuyInfo;
  paper?: PaperBuyInfo;
  rentInfo?: RentInfo;
  pointBackInfo?: PointBackInfo;
}

export interface Property {
  is_adult_only: boolean;
  is_magazine: boolean;
  is_new_book: boolean;
  is_novel: boolean;
  is_open: boolean;
  is_somedeal: boolean;
  is_trial: boolean;
  preview_rate: number;
}

export interface LinkedSeriesBookInfo {
  [b_id: string]: { b_id: string; is_opened: boolean };
}

export interface DeviceSupport {
  android: boolean;
  ios: boolean;
  mac: boolean;
  paper: boolean;
  web_viewer: boolean;
  windows: boolean;
}

export interface SeriesProperty {
  is_comic_hd: boolean;
  is_completed: boolean;
  is_serial: boolean;
  is_serial_complete: boolean;
  is_wait_free: boolean;
  opened_book_count: number;
  opened_last_volume_id: string;
  prev_books: [] | LinkedSeriesBookInfo;
  next_books: [] | LinkedSeriesBookInfo;
  last_volume_id: string;
  title?: string;
  total_book_count: number;
  unit: string | null;
}

export interface Publish {
  ebook_publish: string;
  ridibooks_publish: string;
  ridibooks_register: string;
}

export interface Publisher {
  cp_name: string;
  id: number;
  name: string;
}

export interface Series {
  id: string;
  price_info: SeriesPriceInfo;
  property: SeriesProperty;
  volume: number;
}

export interface ThumbnailInfo {
  large: string;
  small: string;
  xxlarge: string;
}

export interface SetBook {
  member_books_count: number;
}

export interface ClientBookFields {
  isAvailableSelect: boolean;
  isAlreadyCheckedAtSelect: boolean;
}

export interface Book {
  id: string;
  authors: any; // will remove
  authors_ordered: Author[];
  categories: Category[];
  title: Title;
  file: File;
  last_modified: string;
  price_info: PriceInfo;
  property: Property;
  publish: Publish;
  publisher: Publisher;
  series?: Series;
  support: DeviceSupport;
  thumbnail: ThumbnailInfo;
  setbook?: SetBook;

  //
  is_deleted?: boolean;

  // client field
  isAvailableSelect?: boolean;
  thumbnailId?: string; // 시리즈 여부, 완결 여부 판단해서 최종적으로 보여 줄 thumbnail Id

  clientBookFields: ClientBookFields | null;
}

export interface Author {
  name: string;
  id: number;
  role: AuthorRole;
}
