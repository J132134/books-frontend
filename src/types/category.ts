export interface Category {
  id: number;
  name: string;
  parent: number;
  genre_v1: string; // genre?
  genre_v2: string; // genre?
  is_active: boolean;
  usable: boolean;
  visible: boolean;
  use_series: boolean;
  last_modified: string;
}
