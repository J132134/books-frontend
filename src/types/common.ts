import { Store } from 'redux';
import { NextPageContext } from 'next';
import { Request } from 'express';

export interface ConnectedInitializeProps extends NextPageContext {
  store: Store;
  req?: Request;
}
