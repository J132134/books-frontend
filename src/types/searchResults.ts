import { AuthorRole } from 'src/types/book';

export interface Author {
  popular_book_title: string;
  book_count: number;
  name: string;
  id: number;
  highlight: { name?: string };
}

export interface AuthorResult {
  total: number;
  authors: Author[];
}

export interface Aggregation {
  category_id: number;
  category_name: string;
  doc_count: number;
}

export interface Tag {
  tag_name: string;
  tag_id: number;
}
export interface SeriesPriceInfo {
  series_arg: number;
  total_regular_price: number;
  total_price: number;
  type: string; // ex('normal' | ... )
  free_book_count: number;
  series_id: string;
  regular_price: number;
  max_price: number;
  book_count: number;
  min_price: number;
  arg: number;
  min_nonzero_price: number;
  series_discount_rate: number;
}

export interface AuthorsInfo {
  role: AuthorRole | string;
  name: string;
  author_id: number;
  native_name?: string;
  alias_name?: string;
  order: number;
}

export interface HighlightInfo {
  web_title_title?: string; // todo escaping
  sub_title?: string; // todo escaping
  author?: string; // todo escaping
  author2?: string; // todo escaping
  translator?: string; // todo escaping
  publisher?: string; // todo escaping
}

export interface EventInfo {
  event_subgroup_id: number;
  event_id: number;
  event_name: string;
  event_subgroup_name: string;
}

export interface SearchBookDetail {
  buyer_rating_score: number;
  is_series_complete: boolean;
  sub_title: string;
  author: string;
  event_info?: EventInfo[];
  translator: string;
  age_limit: number;
  tags_info: Tag[];
  title: string;
  is_rental: boolean;
  author2: string;
  book_count: number;
  web_title: string;
  b_id: string;
  price: number;
  _score?: number;
  buyer_review_count: number;
  publisher: string;
  buyer_rating_count: number;
  web_title_title: string;
  authors_info: AuthorsInfo[];
  desc: string; // include tags... ex) <b>&lt;책소개&gt;</b>\n\n<strong>“사람들이 나보고 맘충이래.” \r\n\r\n한국에서 여자로 살아가는 일 \r\n그 공포, 피로, 당황, 놀람, 혼란, 좌절의 \r\n연속에 대한 인생 현장 보고서</strong>  \r\n\r\n조남주 장편소설 『82년생 김지영』이 민음사 ‘오늘의 젊은 작가’ 시리즈로 출간되었다. 조남주 작가는 2011년, 지적 장애가 있는 한 소년의 재능이 발견되면서 벌어지는 사건을 통해 삶의 부조리를 현실적이면서도 따뜻하게 그려낸 작품 『귀를 귀울이면』으로 ‘문학동네소설상’을 받으며 데뷔했다. 시사 교양 프로그램에서 10년 동안 일한 방송 작가답게 ';
  series_prices_info: SeriesPriceInfo[];
  highlight: HighlightInfo;
}

export interface BookResult {
  total: number;
  books: SearchBookDetail[];
  aggregations: Aggregation[]; // 책이 포함 된 categories
}

export interface SearchResult {
  author: AuthorResult;
  book: BookResult;
}

export type InstantSearchBookResult = Omit<BookResult, 'total' | 'aggregations'>;
export type InstantSearchAuthorResult = Omit<AuthorResult, 'total'>;

export interface InstantSearchResult {
  book: InstantSearchBookResult;
  author: InstantSearchAuthorResult;
}
