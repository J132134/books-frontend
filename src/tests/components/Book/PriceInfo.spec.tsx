import * as React from 'react';
import { getByText, render } from '@testing-library/react';
import PriceInfo from 'src/components/Search/SearchLandscapeBook/PriceInfo';
import * as SearchTypes from 'src/types/searchResults';
import * as BookApi from 'src/types/book';
import '@testing-library/jest-dom/extend-expect';

import searchApiPriceExistFixture from './fixtures/search-api-price-exist.fixture.json';
import trialBookFixture from './fixtures/trial-book.fixture.json';
import freeBookFixture from './fixtures/free-book.fixture.json';
import priceZeroFixture from './fixtures/price-zero.fixture.json';

interface Fixture {
  searchApiResult: SearchTypes.SearchBookDetail;
  bookApiResult: BookApi.Book;
}

function renderPriceInfo(fixture: Fixture, genre = '') {
  return render(
    <PriceInfo
      searchApiResult={fixture.searchApiResult}
      bookApiResult={fixture.bookApiResult}
      genre={genre}
    />,
  );
}

describe('PriceInfo test', () => {
  test('trial free book render', () => {
    const { container } = renderPriceInfo(trialBookFixture);
    expect(container.textContent).toBe('구매무료 ');
  });

  test('free book render', () => {
    const { container } = renderPriceInfo(freeBookFixture);
    expect(container.textContent).toBe('구매무료 ');
  });

  test('rental, buy price render', () => {
    const { container } = renderPriceInfo(searchApiPriceExistFixture);
    expect(container.textContent).toBe('대여1,900원 구매12,800원 ');
  });
  test('render correct rental & buy price', () => {
    const { container } = renderPriceInfo(priceZeroFixture);
    expect(container.textContent).toBe('구매100원 ');
  });
});
