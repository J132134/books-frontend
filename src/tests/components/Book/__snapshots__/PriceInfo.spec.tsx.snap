// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceInfo test free book render 1`] = `
<div>
  <dl
    class="css-13w561e-PriceItem e18h15o99"
  >
    <dt
      class="css-qvy4wk-PriceTitle-priceBase e18h15o910"
    >
      구매
    </dt>
    <dd>
      <span
        class="css-1jt67wx-Price-priceBase e18h15o911"
      >
        무료
      </span>
       
    </dd>
  </dl>
</div>
`;

exports[`PriceInfo test render correct rental & buy price 1`] = `
<div>
  <dl
    class="css-13w561e-PriceItem e18h15o99"
  >
    <dt
      class="css-qvy4wk-PriceTitle-priceBase e18h15o910"
    >
      구매
    </dt>
    <dd>
      <span
        class="css-1jt67wx-Price-priceBase e18h15o911"
      >
        100원
      </span>
       
    </dd>
  </dl>
</div>
`;

exports[`PriceInfo test rental, buy price render 1`] = `
<div>
  <dl
    class="css-13w561e-PriceItem e18h15o99"
  >
    <dt
      class="css-qvy4wk-PriceTitle-priceBase e18h15o910"
    >
      대여
    </dt>
    <dd>
      <span
        class="css-1jt67wx-Price-priceBase e18h15o911"
      >
        1,900원
      </span>
       
    </dd>
  </dl>
  <dl
    class="css-13w561e-PriceItem e18h15o99"
  >
    <dt
      class="css-qvy4wk-PriceTitle-priceBase e18h15o910"
    >
      구매
    </dt>
    <dd>
      <span
        class="css-1jt67wx-Price-priceBase e18h15o911"
      >
        12,800원
      </span>
       
    </dd>
  </dl>
</div>
`;

exports[`PriceInfo test trial free book render 1`] = `
<div>
  <dl
    class="css-13w561e-PriceItem e18h15o99"
  >
    <dt
      class="css-qvy4wk-PriceTitle-priceBase e18h15o910"
    >
      구매
    </dt>
    <dd>
      <span
        class="css-1jt67wx-Price-priceBase e18h15o911"
      >
        무료
      </span>
       
    </dd>
  </dl>
</div>
`;
